rfcs/20221006-frontend_event_tracking.md

```diff
@@ -0,0 +1,469 @@
+   * on the external sources dropdown.
+   *
+   * This event is mainly used as part of a funnel leading to a
+   * `LOAD_MORE` or `VIEW_EXTERNAL_SOURCES` event.
+   *
+   * Questions:
+   *   - Do users use meta search after reaching the result end?
+   *   - Do users find a result before reaching the end of the results?
+   */
+  REACH_RESULT_END: {
+    /** The media type being searched */
+    mediaType: MediaType;
+    /** The search term */
+    query: string;
+    /** The current page of results the user is on. */
+    resultPage: number;
+  };
+  /**
+   * Whenever a search results in less than a full page
+   * of results or no results.
+   *
+   * Questions:
+   *   - How often do searches provide limited or no results?
+   *   - Do users visit external sources when results are insufficient?
+   *   - What do users do after a search has no results?
+   */
+  INSUFFICIENT_RESULTS: {
+    /** The media type being searched */
+    mediaType: MediaType;
+    /** The search term */
+    query: string;
+    /** The number of results returned */
+    resultCount: number;
+  };
+  /**
+   * Description: Whenever the user clicks the load more button
+   * Questions:
+   *   - On what page do users typically find a result?
+   *   - How often and how many pages of results do users load?
+   *   - Can we experiment with the types of results / result rankings
+   *     on certain pages, pages that users don't usually choose a result
+   *     from anyway?
+   */
+  LOAD_MORE_RESULTS: {
+    /** The media type being searched */
+    mediaType: MediaType;
+    /** The search term */
+    query: string;
+    /** The current page of results the user is on. */
+    resultPage: number;
+  };
+  /**
+   * Description: When a user opens the external sources popover.
+   * Questions:
+   *   - How often do users use this feature?
+   *   - Under what conditions to users use this feature? No results?
+   *     Many results, but none they actually select?
+   */
+  VIEW_EXTERNAL_SOURCES: {
+    /** The media type being searched */
+    mediaType: MediaType;
+    /** The search term */
+    query: string;
+    /** The current page of results the user is on. */
+    resultPage: number;
+  };
+  /**
+   * Description: When the user chooses an external source from the dropdown of external sources
+   * Questions:
+   *   - Which external sources are most popular? This could drive inclusion in Openverse.
+   *   - Are certain media types more popular externally?
+   */
+  SELECT_EXTERNAL_SOURCE: {
+    /** The name of the external source */
+    name: string;
+    /** The full URL of the source */
+    url: string;
+    /** The media type being searched */
+    mediaType: MediaType;
+    /** The search term */
+    query: string;
+  };
+  /**
+   * Description: The user opens the menu which lists pages.
+   * Questions:
+   *   - How often is this menu used?
+   *   - Is this menu visible enough?
+   */
+  OPEN_PAGES_MENU: {};
```

Comment:
Noting here that this will mostly track mobile users because on desktop the pages menu is always open.

Reply:
Good point! I conceived of this for the old header design.